apps/web/src/components/borrow/modal/DepositModal.tsx

```diff
@@ -0,0 +1,263 @@
+        const approvedTokens = await Tokens.getAllowance(
+          tokenContract,
+          currentAccount,
+          'deposit',
+        )
+        setShouldApproveContract(approvedTokens.toString() === '0')
+        setHasError(null)
+      } catch (error) {
+        setHasError(error)
+      }
+    }
+  }, [currentAccount, tokenContract])
+
+  useEffect(() => {
+    if (isSupportedNetwork) {
+      void checkAllowance()
+    }
+  }, [checkAllowance, isSupportedNetwork])
+
+  const getHealthFactor = async () => {
+    setLoading(true)
+    try {
+      const response = await BorrowAndLending.lendingPool.getUserAccountData(
+        lendingPoolContract,
+        currentAccount,
+      )
+      setHealthFactor(Number(ethers.utils.formatUnits(response.healthFactor)))
+    } catch (error) {
+      setHasError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+  useEffect(() => {
+    void getHealthFactor()
+  }, [])
+
+  const approve = async () => {
+    setLoading(true)
+    try {
+      const response = await Tokens.approveToken(tokenContract, 'deposit')
+      const isApproved = await response.wait()
+
+      setApproved(!!isApproved)
+    } catch (error) {
+      setHasError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  const deposit = async (amount: number) => {
+    setLoading(true)
+    try {
+      const response = await BorrowAndLending.lendingPool.deposit(
+        lendingPoolContract,
+        reserveData.underlyingAsset,
+        amount,
+        currentAccount,
+      )
+      const isDeposited = await response.wait()
+      setDepositCompleted(!!isDeposited)
+    } catch (error) {
+      setHasError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  const depositValidationSchemaa = Yup.object().shape({
+    amount: Yup.number()
+      .typeError(t('global.errors.numbersOnly'))
+      .positive(t('global.errors.positiveValue'))
+      .max(Number(reserveData.balance), t('global.errors.exceededBalance'))
+      .required(t('global.errors.required')),
+  })
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <Formik
+        initialValues={{ amount: '' }}
+        validationSchema={depositValidationSchemaa}
+        onSubmit={values => deposit(Number(values.amount))}
+      >
+        {({
+          values,
+          errors,
+          touched,
+          handleChange,
+          handleBlur,
+          handleSubmit,
+          setFieldValue,
+        }) => {
+          const isDepositDisabled =
+            !isSupportedNetwork ||
+            loading ||
+            !!errors.amount ||
+            !values.amount ||
+            hasError ||
+            (shouldApproveContract && !approved)
+
+          return (
+            <form onSubmit={handleSubmit}>
+              <StackLayout gap={5}>
+                <StackLayout gap={3}>
+                  <TableInput
+                    id="amount"
+                    name="amount"
+                    type="number"
```

Comment:
we can even add min={0} max={Number(reserveData.balance)} It's not bullet-proof (meaning we should still have our validation), but it's nice to have.